_ingest-pipelines/conditional-pipeline-processor.md

```diff
@@ -0,0 +1,222 @@
+---
+layout: default
+title: Conditionals with the pipeline processor
+nav_order: 60
+---
+
+# Conditionals with the pipeline processor
+
+The pipeline processor in ingest pipelines allows conditional execution of different sub-pipelines based on document contents. This provides powerful flexibility when different types of documents require separate processing logic. You can use the `if` parameter in the `pipeline` processor to direct documents into different pipelines based on field values, data types, or content structure. Each pipeline can then apply its own set of processors independently. This approach keeps your pipelines modular and maintainable by applying logic only where it’s relevant.
+
+## Example: Route logs by service
+
+The following example shows how to route logs to different sub-pipelines depending on the `service.name` field in the document.
+
+Create the first pipeline name `webapp_logs`:
+
+```json
+PUT _ingest/pipeline/webapp_logs
+{
+  "processors": [
+    { "set": { "field": "log_type", "value": "webapp" } }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+Create the second pipeline name `api_logs`:
+
+```json
+PUT _ingest/pipeline/api_logs
+{
+  "processors": [
+    { "set": { "field": "log_type", "value": "api" } }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+Create the main routing pipeline name `service_router` that routes the documents to the corresponding pipelines based on `service.name`:
+
+```json
+PUT _ingest/pipeline/service_router
+{
+  "processors": [
+    {
+      "pipeline": {
+        "name": "webapp_logs",
+        "if": "ctx.service?.name == 'webapp'"
+      }
+    },
+    {
+      "pipeline": {
+        "name": "api_logs",
+        "if": "ctx.service?.name == 'api'"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+Use the following request to simulate the pipelines:
+
+```json
+POST _ingest/pipeline/service_router/_simulate
+{
+  "docs": [
+    { "_source": { "service": { "name": "webapp" }, "message": "Homepage loaded" } },
+    { "_source": { "service": { "name": "api" }, "message": "GET /v1/users" } },
+    { "_source": { "service": { "name": "worker" }, "message": "Task started" } }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+The returned documents demonstrate the following:
```

Comment:
```suggestion
The response confirms that the first document was processed by the `webapp_logs` pipeline and the second document was processed by the `api_logs` pipeline. The third document remains unchanged because it doesn't match any conditions:
```